### DISCUSSION TITLE
Setting up trilium docker server access from anywhere

### DISCUSSION THREAD

**Discussion by prishur** [TriliumNext/Trilium#3006] [Q&A] [answered]
I am trying to have a setup where a docker container is running on one machine and a couple desktop clients (from different or same machine) sync to the server. I ran the command mentioned in the [wiki](https://github.com/zadam/trilium/wiki/Docker-server-installation#available-anywhere):
`docker run -d -p 0.0.0.0:8080:8080 -v ~/trilium-data:/home/node/trilium-data zadam/trilium:[VERSION] `

How does a client now connect to this container over HTTP? In other words, what server  address do I provide in the sync configuration of the client? My understanding so far is that I'd need to attach the running docker container to a pulic host name and use that hostname as server address but couldn't find instructions for that in the wiki or discussions. Is there a simpler way to sync across multiple devices ? 

I understand using HTTPS and setting up certificates would also be needed, but before that I'm trying to have it woking over HTTP.

Update: By using `docker inspect <Container ID>`, I was able to see an IP address for the container. It would be super helpful if someone can share:
 1.  how to make this port accessible over internet.
 2. how to generate a TLS certificate for this address (I looked up certbot but i requiers a domain name for generating a cert. Not sure what domain name to use.)

**Comment by FliegendeWurst** [comment, marked as answer]
> In other words, what server address do I provide in the sync configuration of the client?

If the client and server are in the same LAN, you can use the hostname of the server (see `/etc/hostname`). The port is determined by the docker command: `[...] -p 0.0.0.0:12345:8080 [...]` would open port 12345.  
If you need to access the sync server over the internet, the setup will be more involved.. are you running your sync server in the cloud?

> Is there a simpler way to sync across multiple devices ?

https://trilium.cc/ is the simplest option.

**Reply by prishur** [reply]
> > In other words, what server address do I provide in the sync configuration of the client?
> 
> If the client and server are in the same LAN, you can use the hostname of the server (see `/etc/hostname`). The port is determined by the docker command: `[...] -p 0.0.0.0:12345:8080 [...]` would open port 12345. If you need to access the sync server over the internet, the setup will be more involved.. are you running your sync server in the cloud?
> 
> > Is there a simpler way to sync across multiple devices ?
> 
> https://trilium.cc/ is the simplest option.

I'm running the sync server on a personal device for now. Wouldn't mind doing more setup to get it working, if it means I'll be learning something about docker, networks and security in the process. Would be awesome if you could outline the steps to sync over internet.

**Reply by FliegendeWurst** [reply]
You could try setting up a VPN on your home router (precise setup depends on your router model and ISP). Then connect all devices to the VPN. If everything is setup correctly, your sync server should be reachable on every client using the local hostname (`/etc/hostname`) or IP.

**Reply by prishur** [reply]
Thanks

> You could try setting up a VPN on your home router (precise setup depends on your router model and ISP). Then connect all devices to the VPN. If everything is setup correctly, your sync server should be reachable on every client using the local hostname (`/etc/hostname`) or IP.

Thanks, that's super useful. I'm curious though: If I switch to a cloud sync server, what additional setup would be needed.

**Reply by FliegendeWurst** [reply]
Additional setup would be setting up a reverse proxy (HTTPS access -> local HTTP request to Trilium sync server). You can get a TLS certificate from [Let's Encrypt](https://letsencrypt.org/). You'll also need a domain (any free dynamic DNS works).
